sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -648,13 +649,80 @@ public override void Dispose()
             }
         }
 
+        private async Task<AmqpSession> CreateSessionIfNeededAsync(AmqpConnection connection, TimeSpan timeout, string transactionGroupKey)
+        {
+            AmqpSession session;
+            FaultTolerantAmqpObject<Controller> faultTolerantController = null;
+            if (transactionGroupKey != null)
+            {
+                if (TransactionGroups.ContainsKey(transactionGroupKey))
```

Comment:
I'd recommend `TryGetValue` here, otherwise this is a race condition.   They key may be present in the `if` check and removed before line 660 executes.

Reply:
Well it wouldn't be a race condition because nothing is removed from the dictionary, but it is a fair point.